src/lib/non_zero_curve_point/non_zero_curve_point.ml

```diff
@@ -62,6 +65,11 @@ module Compressed = struct
       let sexp_of_t t = to_base58_check t |> Sexp.of_string
 
       let t_of_sexp sexp = Sexp.to_string sexp |> of_base58_check_exn
+
+      let gen =
```

Comment:
This can generate invalid curve points. It doesn't matter in these tests, but to avoid a future gotcha, use the generator for uncompressed curve points and then compress them.

Reply:
I'll fix it, to avoid any confusion